### PR TITLE
Refactor SerializedPage checksum

### PR DESCRIPTION
**Pull request by arhimondr**
Avoid sending a hostname with every page. The remote host is known in
ExchangeClient.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by mbasmanova — APPROVED** [review]
Looks good to me.

**Review by bhhari — APPROVED** [review]
looks good

File: presto-spi/src/main/java/com/facebook/presto/spi/page/PageDataOutput.java [hunk 1]
```diff
@@ -34,7 +35,7 @@ public PageDataOutput(SerializedPage serializedPage)
     @Override
     public long size()
     {
-        return SIZE_OF_INT * 3 + SIZE_OF_BYTE + serializedPage.getSizeInBytes();
+        return SIZE_OF_INT * 3 + SIZE_OF_BYTE + SIZE_OF_LONG + serializedPage.getSizeInBytes();
```
Comment (bhhari) [inline presto-spi/src/main/java/com/facebook/presto/spi/page/PageDataOutput.java:38]: can we use a constant for the metadata size to PAGE_METADATA_SIZE, which is also in SerializedPageWriteListener

File: presto-main/src/main/java/com/facebook/presto/server/SerializedPageWriteListener.java [hunk 2]
```diff
@@ -78,6 +79,8 @@ public void onWritePossible()
                 bufferPosition += SIZE_OF_INT;
                 slice.setInt(bufferPosition, page.getSizeInBytes());
                 bufferPosition += SIZE_OF_INT;
+                slice.setLong(bufferPosition, page.getChecksum());
```
Comment (bhhari) [inline presto-main/src/main/java/com/facebook/presto/server/SerializedPageWriteListener.java:82]: @arhimondr this is what got missed in the previous PR. Wondering if we can reuse the logic from PageSerdeUril.writeSerializedPage

File: presto-spi/src/main/java/com/facebook/presto/spi/page/PageDataOutput.java (hunk 1 above)
Comment (arhimondr) [reply presto-spi/src/main/java/com/facebook/presto/spi/page/PageDataOutput.java:38]: Good point. I had to move the constant to the `PagesSerdeUtil` though to make it available in both modules.

File: presto-main/src/main/java/com/facebook/presto/server/SerializedPageWriteListener.java (hunk 2 above)
Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/server/SerializedPageWriteListener.java:82]: Yeah, let me refactor it

**Review by bhhari — COMMENTED** [review]
thanks for addressing the comments @arhimondr. LGTM

**Pull request merged**